### PR TITLE
CDD-2032_online_proc

### PR DESCRIPTION
**Pull request by paulblakehmrc**

### CODE REVIEW

File: test/integration/CustomsNotificationMetricsConnectorSpec.scala [hunk 1]
```diff
@@ -79,31 +79,31 @@ with BeforeAndAfterAll with CustomsNotificationMetricsService with AuditService
       intercept[RuntimeException](await(sendValidRequest())).getCause.getClass shouldBe classOf[NotFoundException]
```
Comment (googley42) [inline test/integration/CustomsNotificationMetricsConnectorSpec.scala]: blank line after `intercept ..` pls to preserve `GIVEN/WHEN/THEN` block formatting and also for the other tests

File: test/util/TestData.scala [hunk 2]
```diff
@@ -22,14 +22,16 @@ import java.util.UUID
...
 import play.api.libs.json.{JsValue, Json}
 import play.api.mvc.AnyContentAsXml
 import play.api.test.FakeRequest
+import play.mvc.Http.MimeTypes
+import reactivemongo.bson.BSONObjectID
 import uk.gov.hmrc.customs.notification.controllers.CustomHeaderNames._
-import uk.gov.hmrc.customs.notification.controllers.CustomMimeType
+import uk.gov.hmrc.customs.notification.controllers.{CustomMimeType, RequestMetaData}
 import uk.gov.hmrc.customs.notification.domain._
 import uk.gov.hmrc.customs.notification.util.DateTimeHelpers._
```
Comment (googley42) [inline test/util/TestData.scala]: unused import `import uk.gov.hmrc.workitem.{ToDo, WorkItem}`

File: app/uk/gov/hmrc/customs/notification/connectors/PushNotificationServiceWorkItemConnector.scala [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import play.api.http.Status.NO_CONTENT
+import uk.gov.hmrc.customs.api.common.config.ServiceConfigProvider
+import uk.gov.hmrc.customs.notification.domain.{PushNotificationRequest, PushNotificationRequestBody}
+import uk.gov.hmrc.customs.notification.logging.NotificationLogger
+import uk.gov.hmrc.http.{HeaderCarrier, HttpException, HttpResponse}
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
```
[the remark this answers is not in the document]
Comment (paulblakehmrc) [reply app/uk/gov/hmrc/customs/notification/connectors/PushNotificationServiceWorkItemConnector.scala]: ok

File: app/uk/gov/hmrc/customs/notification/controllers/customsNotificationControllers.scala [hunk 4]
```diff
@@ -101,4 +105,33 @@ class CustomsNotificationController @Inject()(logger: NotificationLogger,
...
+                                                          configService: CustomsNotificationConfig,
+                                                          dateTimeService: DateTimeService)
+  extends CustomsNotificationController(logger, customsNotificationService, callbackDetailsConnector, configService, dateTimeService) {
+
+  def handleNotification(xml: NodeSeq, md: RequestMetaData, apiSubscriptionFieldsResponse: ApiSubscriptionFieldsResponse)(implicit hc: HeaderCarrier): Future[Boolean] = {
+    customsNotificationService.handleNotification(xml, md)
+  }
+
 }
+
```
[the remark this answers is not in the document]
Comment (paulblakehmrc) [reply app/uk/gov/hmrc/customs/notification/controllers/customsNotificationControllers.scala]: Done

File: conf/application.conf [hunk 5]
```diff
@@ -94,6 +94,7 @@ pull.exclude.email.delay.duration.seconds = 0
 pull.exclude.older.milliseconds = 5000
 pull.exclude.csIds.0 = 99999999-9999-9999-9999-999999999999
```
[the remark this answers is not in the document]
Comment (paulblakehmrc) [reply conf/application.conf]: done

**Pull request merged**